server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -308,7 +307,6 @@ public ProductDTO addBatchContent(
             throw new ForbiddenException(i18n.tr("product \"{0}\" is locked", product.getId()));
         }
 
-        this.productCurator.lock(product);
```

Comment:
I'm a little concerned about removing these locks here. Admittedly, having the resource lock the product is a bit of a hack anyway, so something should be done to improve this; but the scenario where two update requests for the same product at the same time worries me. Feels like we'll end up with a typical race condition where both succeed but one update is clobbered.